### PR TITLE
S3 uploads

### PR DESCRIPTION
**Pull request by benjaminpaige**
## Purpose

Allow for upload / download / delete of files in Upload comp

#### Linked Issues to Close

https://qmacbis.atlassian.net/browse/OY2-15240

#### Approach

When a user uploads a file it is immediately tagged with virusScanStatus = 'PENDING'

once uploaded the avscan lambda scans the file and retags with either 'CLEAN' 'INFECTED' or 'ERROR'

if the file is tagged with 'CLEAN' or 'PENDING' download by user is allowed. 

A progress indicator has been added for upload status. File name is clickable for file download, and 'X' button has been added for deletions by user. Deleted files are removed from s3

When files are added/removed the form/measure is automatically saved. 

#### Possible next steps

It might be nice to have a process regularly checking for infected files and removing them (or handling them in some way) or remove them if/when they are found to be infected.

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/uploads/serverless.yml [hunk 1]
```diff
@@ -28,6 +28,14 @@ provider:
...
             - dynamodb:UpdateItem
             - dynamodb:DeleteItem
+            - s3:GetObject
+            - s3:GetObjectTagging
+            - s3:PutObject
+            - s3:PutObjectAcl
+            - s3:PutObjectTagging
+            - s3:PutObjectVersionTagging
+            - s3:DeleteObject
+            - s3:ListBucket
```
Comment (13bfrancis) [inline services/uploads/serverless.yml:38]: Great job figuring out which permissions to add!

**Review by 13bfrancis — DISMISSED** [review]

**Review by jdinh8124 — DISMISSED** [review]
From what I can gather it looks good

**Review by abelzer212 — APPROVED** [review]

**Review by abelzer212 — APPROVED** [review]

File: services/uploads/serverless.yml (hunk 1 above)
Comment (benjaminpaige) [inline services/uploads/serverless.yml:38]: will we need to add these permissions in master/val/prod @abelzer212 @jdinh8124

Comment (abelzer212) [reply services/uploads/serverless.yml:38]: What do you mean by that? As in, do we need to make a CMS ticket? Or Do we need to include them in this PR?

Comment (abelzer212) [reply services/uploads/serverless.yml:38]: I do think they should be in this PR

**Pull request merged**